File: 275. H-Index II.py
class Solution(object):
    def hIndex(self, citations):
        """
        :type citations: List[int]
        :rtype: int
        """
        start = 0
        end = len(citations)
        while start < end:
            mid = (start + end) >> 1
            if citations[mid] >= len(citations) - mid:
                end = mid
            else:
                start = mid +1
        return len(citations) - start

File: 275. H-Index II.cpp
class Solution {
public:
    int hIndex(vector<int>& citations) {
        int end = citations.size();
        int start = 0;
        while (start < end)
        {
            int mid = (end + start) >> 1;
            if (citations[mid] >= citations.size() - mid)
                end = mid;
            else
                start = mid + 1;
        }
        return citations.size() - start;
    }
};
